component-studio/component-runtime-di/src/main/java/org/talend/sdk/component/runtime/di/studio/RuntimeContextInjector.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (C) 2006-2022 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.runtime.di.studio;
+
+import java.util.Map;
+
+import org.talend.sdk.component.api.context.RuntimeContext;
+import org.talend.sdk.component.runtime.base.Delegated;
+import org.talend.sdk.component.runtime.base.Lifecycle;
+import org.talend.sdk.component.runtime.manager.ComponentManager;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class RuntimeContextInjector {
+
+    /**
+     * Inject runtime context object to runtime input/processor/standalone runner object
+     */
+    public static void injectLifecycle(final Lifecycle lifecycle, final Map<String, Object> runtimeContext) {
+        if (lifecycle instanceof Delegated) {
+            final Object delegate = ((Delegated) lifecycle).getDelegate();
+
+            Class<?> currentClass = delegate.getClass();
+            while (currentClass != null && currentClass != Object.class) {
+                ReflectionUtils.findFields(currentClass, RuntimeContext.class).forEach(f -> {
```

Comment:
Yes, now we are handling all classes in the hierarchy of the instance: "delegate". Before that, we on each iteration over and over get the same exact class of the delegate object and its parent classes weren't searched for the fields.

Reply:
Please, look at the comment in the ReflectionUtils.findFields(currentClasss, RuntimeContext.class) method. 
Maybe we need to call its logic inside "callInLoader" too.